test/functional/apps/dashboard/esql_controls/field_control.ts

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import expect from '@kbn/expect';
+
+import { FtrProviderContext } from '../../../ftr_provider_context';
+
+export default function ({ getService, getPageObjects }: FtrProviderContext) {
+  const retry = getService('retry');
+  const kibanaServer = getService('kibanaServer');
+  const { dashboard, timePicker, common } = getPageObjects(['dashboard', 'timePicker', 'common']);
+  const testSubjects = getService('testSubjects');
+  const monacoEditor = getService('monacoEditor');
+  const dashboardAddPanel = getService('dashboardAddPanel');
+  const browser = getService('browser');
+  const comboBox = getService('comboBox');
+  const elasticChart = getService('elasticChart');
+
+  describe('dashboard - add a field type ES|QL control', function () {
+    before(async () => {
+      await kibanaServer.savedObjects.cleanStandardList();
+      await kibanaServer.importExport.load(
+        'test/functional/fixtures/kbn_archiver/dashboard/current/kibana'
+      );
+      await kibanaServer.uiSettings.replace({
+        defaultIndex: '0bf35f60-3dc9-11e8-8660-4d65aa086b3c',
+      });
+    });
+
+    after(async () => {
+      await dashboard.navigateToApp();
+      await testSubjects.click('discard-unsaved-New-Dashboard');
+    });
+
+    it('should add an ES|QL field control', async () => {
+      await dashboard.navigateToApp();
+      await dashboard.clickNewDashboard();
+      await timePicker.setDefaultDataRange();
+      await dashboard.switchToEditMode();
+      await dashboardAddPanel.clickEditorMenuButton();
+      await dashboardAddPanel.clickAddNewPanelFromUIActionLink('ES|QL');
+      await dashboard.waitForRenderComplete();
+
+      await retry.try(async () => {
+        const panelCount = await dashboard.getPanelCount();
+        expect(panelCount).to.eql(1);
+      });
+
+      await monacoEditor.waitCodeEditorReady('InlineEditingESQLEditor');
+
+      await retry.waitFor('control flyout to open', async () => {
+        await monacoEditor.setCodeEditorValue(''); // clear the default query
```

Comment:
What are your thoughts about moving some of these into an `ESQLControl` service to avoid some of the duplication between the tests?

Reply:
Sure done https://github.com/elastic/kibana/pull/202875/commits/bf64492f119fab7a79984c14361cb5827ff043e0. I can re-use them to other FTs too but I will do it on a follow up PR